### ISSUE TITLE
Secure access to official Kubernetes GCR registries to store capz images

### ISSUE THREAD

**Issue by justaugustus** [kubernetes-sigs/cluster-api-provider-azure#118]
### Status

- [x] (https://github.com/kubernetes/k8s.io/pull/373) Configure GCR staging repo
- [x] (https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/281) Configure cloudbuild.yaml and Makefile to support image-pushing (this will land once we land `v1alpha2` on `master`)
- [x] (https://github.com/kubernetes/test-infra/pull/14440) Configure [image-pushing job](https://github.com/kubernetes/test-infra/tree/master/config/jobs/image-pushing) for capz

---

/kind feature

**Describe the solution you'd like**
Currently, we use a personal container registry of mine (https://quay.io/repository/k8s/cluster-api-azure-controller?tab=tags) to store both production and development container images for capz.

We should move this to Kubernetes project-supported repositories.
There is an active effort from the k8s-infra-team to create these, so this is on hold until we have movement on that side.

**Anything else you would like to add:**
ref: https://github.com/kubernetes/k8s.io/issues/158#issuecomment-462047593
> ON Friday Feb 8th Tim and i ended up setting up the following for GCR.
> 
> As a first step, we are setting up staging repositories.
> 
> * gcr.io/k8s-staging-csi
> * gcr.io/k8s-staging-cluster-api
> * gcr.io/k8s-staging-coredns
> 
> Folks in the following google groups will be able to push to the staging repository:
> 
> * [k8s-infra-gcr-staging-csi@googlegroups.com](mailto:k8s-infra-gcr-staging-csi@googlegroups.com)
> * [k8s-infra-gcr-staging-cluster-api@googlegroups.com](mailto:k8s-infra-gcr-staging-cluster-api@googlegroups.com)
> * [k8s-infra-gcr-staging-coredns@googlegroups.com](mailto:k8s-infra-gcr-staging-coredns@googlegroups.com)
> 
> The following google group has the admin privileges over all the GCR repos.
> 
> * [k8s-infra-gcr-admins@googlegroups.com](mailto:k8s-infra-gcr-admins@googlegroups.com)
> 
> Notes:
> 
> * Only release artifacts should be uploaded here
> * artifacts will be cleaned up periodically (older than 2 weeks?) to prevent the use of these repos beyond staging
> * We will have additional workflow for promotion from here to the main k8s.gcr.io repo
> * We will be adding sig leads to the google groups sometime next week to kick the tires
> 
> cc @thockin @hh
> 
> PS: next on the list is cloud-provider(s)

/assign

**Comment by justaugustus** [comment]
This is complete, based on the instructions for image pushing: https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/README.md
Push job status can be viewed here: https://testgrid.k8s.io/sig-cluster-lifecycle-image-pushes#post-cluster-api-provider-azure-push-images

/close
